core/src/ldml/ldml_processor.cpp

```diff
@@ -467,23 +468,26 @@ void
 ldml_event_state::emit_text( km_core_usv ch) {
   assert(ch != LDML_UC_SENTINEL);
   state->context().push_character(ch);
-  state->actions().push_character(ch);
+  text.push_back(ch);
 }
 
 void
 ldml_event_state::emit_marker( KMX_DWORD marker_no) {
   assert(km::core::kmx::is_valid_marker(marker_no));
-  state->actions().push_marker(marker_no);
+  // No markers in the action struct!
   state->context().push_marker(marker_no);
 }
 
 void ldml_event_state::emit_invalidate_passthrough_keystroke() {
   if ((vk < 0x100) && km::core::kmx::vkey_to_contextreset[vk]) {
-    state->actions().push_invalidate_context();
+    // TODO-LDML: how to invalidate context?
+    context_clear();
```

Comment:
Engine decides whether a passed-through keystroke is a context invalidation event. So we don't have to make that decision in each keyboard processor, esp. as the set of invalidating keys may vary from platform to platform.

Reply:
OK. Then I'm back to not being able to test that.